### ISSUE TITLE
NotThrowAsync is messing with SynchronizationContext.Current

### ISSUE THREAD

**Issue by TomKuhn** [fluentassertions/fluentassertions#2230]
### Description

We are using Nito AsyncEx to create a single threaded SynchronizationContext, and is active in Current.

Calling

await job.Invoking(f => f.RunAsync()).Should().NotThrowAsync()

causes our RunAsync to be called but with a null SynchronizationContext, this is unexpected to me. 
We are using FluentAssertions v 5.10.3, we have not upgraded to 6.x yet.

Before NotThrowAsync()
<img width="645" alt="image" src="https://github.com/fluentassertions/fluentassertions/assets/3392269/e6886de4-1433-4b4d-90c7-e071087642d8">

Inside Invoked method:

<img width="565" alt="image" src="https://github.com/fluentassertions/fluentassertions/assets/3392269/9bcc2b84-7259-47d9-8bd6-da6fb79db623">

### Reproduction Steps

```csharp

 public sealed class Subject
{
public SynchronizationContext Capture { get; private set; }
public Task DoSomethingAsync()
{
	Capture = SynchronizationContext.Current;
	return Task.CompletedTask;
}
}

[Test]
public void null_sync_context()
{
var subject = new Subject();
AsyncContext.Run(async () =>
{
	await subject.Invoking(f => f.DoSomethingAsync()).Should().NotThrowAsync();
});

subject.Capture.Should().NotBeNull(); //I would not expect Fluent to be messing with my Sync Context!
}

### Expected behavior

SynchronizationContext.Current is left untouched when calling NotThrowAsync();

### Actual behavior

SynchronizationContext.Current is set to null before invoking the callback

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

_No response_

### Other information

_No response_

**Comment by TomKuhn** [comment]
Looking at the code for 6.x, it looks like nothing touches SyncContext any more, and it just seems to directly invoke the Func<Task>. I will try to upgrade

**Comment by lg2de** [comment]
Yes, this was fixed with #1324 released in v6.0.
